ironfish/src/assets/assetsVerifier.ts

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { createRootLogger, Logger } from '../logger'
+import { ErrorUtils } from '../utils'
+import { SetIntervalToken } from '../utils'
+import { AssetsVerificationApi, VerifiedAssets } from './assetsVerificationApi'
+
+export type AssetVerification = {
+  status: 'verified' | 'unverified' | 'unknown'
+}
+
+export class AssetsVerifier {
+  private readonly REFRESH_INTERVAL = 6 * 60 * 60 * 1000 // 6 hours
+
+  private readonly logger: Logger
+  private readonly api: AssetsVerificationApi
+
+  private started: boolean
+  private refreshToken?: SetIntervalToken
+  private verifiedAssets?: VerifiedAssets
+
+  constructor(options?: { apiUrl?: string; logger?: Logger }) {
+    this.logger = options?.logger ?? createRootLogger()
+    this.api = new AssetsVerificationApi({ url: options?.apiUrl })
+    this.started = false
+  }
+
+  start(): void {
+    if (this.started) {
+      return
+    }
+
+    this.started = true
+    void this.refreshLoop()
+  }
+
+  stop(): void {
+    if (!this.started) {
+      return
+    }
+
+    this.started = false
+
+    if (this.refreshToken) {
+      clearTimeout(this.refreshToken)
+    }
+  }
+
+  private async refreshLoop(): Promise<void> {
+    await this.refresh()
+
+    this.refreshToken = setTimeout(() => {
+      void this.refreshLoop()
+    }, this.REFRESH_INTERVAL)
+  }
+
+  private async refresh(): Promise<void> {
+    try {
+      if (this.verifiedAssets) {
+        await this.api.refreshVerifiedAssets(this.verifiedAssets)
+      } else {
+        this.verifiedAssets = await this.api.getVerifiedAssets()
+      }
+    } catch (error) {
+      this.logger.error(
```

Comment:
nit: this is non fatal to the node experience so it should be a warn. If later the source becomes available it'll just pick back up.